File: 145/Solution.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function helper(root:TreeNode | null, save: number[]){
    if(root!=null){
        if(root.left!=null){
            helper(root.left,save);
        }
        if(root.right!=null){
            helper(root.right,save);
        }
        save.push(root.val);
    }
}

function postorderTraversal(root: TreeNode | null): number[] {
    let save = new Array();
    helper(root,save);
    return save;
};

File: 145/Solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func helper(root *TreeNode,save *[]int) {
    if(root!=nil){
        if(root.Left!=nil){
            helper(root.Left,save);
        }
        if(root.Right!=nil){
            helper(root.Right,save);
        }
        *save = append(*save,root.Val);
    }
}
func postorderTraversal(root *TreeNode) []int {
    save := []int{}
    helper(root,&save);
    return save;
}